rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -4402,26 +4413,27 @@ public static Scriptable newCatchScope(
             if (errorObject instanceof NativeError) {
                 ((NativeError) errorObject).setStackProvider(re);
             }
-
-            if (javaException != null && isVisible(cx, javaException)) {
-                Object wrap = cx.getWrapFactory().wrap(cx, scope, javaException, null);
-                ScriptableObject.defineProperty(
-                        errorObject,
-                        "javaException",
-                        wrap,
-                        ScriptableObject.PERMANENT
-                                | ScriptableObject.READONLY
-                                | ScriptableObject.DONTENUM);
-            }
-            if (isVisible(cx, re)) {
-                Object wrap = cx.getWrapFactory().wrap(cx, scope, re, null);
-                ScriptableObject.defineProperty(
-                        errorObject,
-                        "rhinoException",
-                        wrap,
-                        ScriptableObject.PERMANENT
-                                | ScriptableObject.READONLY
-                                | ScriptableObject.DONTENUM);
+            if (ScriptRuntime.isLiveConnectEnabled(scope)) {
+                if (javaException != null && isVisible(cx, javaException)) {
```

Comment:
do not declare these additional properties, when the scope does not support liveConnect